install/kubernetes/cilium/charts/nodeinit/templates/daemonset.yaml

```
@@ -85,7 +85,7 @@ spec:
                     systemctl restart kubelet
 {{- end }}
 
-{{- if .Values.global.gke.enabled }}
+{{- if (or (and .Values.global.gke.enabled .Values.global.masquerade) (and .Values.global.gke.enabled .Values.global.gke.disableDefaultSnat))}}
```

nit: can be as per below, feel free to ignore

```suggestion
{{- if (and ( .Values.global.gke.enabled (or .Values.global.masquerade .Values.global.gke.disableDefaultSnat)))}}
```

---

done